lib/galaxy/managers/workflows.py

```diff
@@ -320,6 +323,31 @@ def attach_stored_workflow(self, trans, workflow):
                 trans.sa_session.commit()
             return stored_workflow
 
+    def get_workflow_by_trs_id_and_version(self, sa_session, trs_id: str, trs_version: str, user_id: int) -> Optional[model.Workflow]:
+        def to_json(column, keys: List[str]):
+            if sa_session.bind.dialect.name == "postgresql":
+                cast = func.cast(func.convert_from(column, "UTF8"), JSONB)
+                for key in keys:
+                    cast = cast.__getitem__(key)
+                return cast.astext
+            else:
+                for key in keys:
+                    column = column.__getitem__(key)
+                return column
+
+        return sa_session.execute(
+            select([model.Workflow])
```

Comment:
```suggestion
            select(model.Workflow)
```
not an error, but deprecated